### ISSUE TITLE
'tensorflow_core.python' not found in tensorflow 1.15.0

### ISSUE THREAD

**Issue by gcervantes8** [pyinstaller/pyinstaller#5002]
## Description of the issue
When importing tensorflow 1.15.0 and running pyinstaller, it gives a module not found error.
Error:  ModuleNotFoundError: No module named 'tensorflow_core.python'

Running the script using python without pyinstaller works.  And running with tensorflow 1.14 works.

## Context information (for bug reports)
* Python 3.7.8 
* Windows 10 (64 bit)
* Output of `pyinstaller --version`: 4.0.dev0

### A minimal example program which shows the error

This is what I'm running (I create a new virtual environment for this test)

```
mkdir temp_venv
python -m venv ./temp_venv
CALL .\temp_venv\Scripts\activate
pip install tensorflow-cpu==1.15.0
pip install https://github.com/pyinstaller/pyinstaller/archive/develop.zip
pyinstaller --noupx --additional-hooks-dir=hooks tf_pyinstaller_test.py
dist\__init__\__init__.exe
```

The tf_pyinstaller_test.py file:

```
import tensorflow

print('Worked!')
```

### Stacktrace / full error message

```
Error traceback:
Traceback (most recent call last):
  File "tf_pyinstaller_test.py", line 1, in <module>
    import tensorflow
  File "c:\users\jerry c\desktop\built\temp_venv\lib\site-packages\PyInstaller\loader\pyimod03_importers.py", line 493, in exec_module
    exec(bytecode, module.__dict__)
  File "tensorflow\__init__.py", line 99, in <module>
  File "c:\users\jerry c\desktop\built\temp_venv\lib\site-packages\PyInstaller\loader\pyimod03_importers.py", line 493, in exec_module
    exec(bytecode, module.__dict__)
  File "tensorflow_core\__init__.py", line 28, in <module>
  File "tensorflow\__init__.py", line 50, in __getattr__
  File "tensorflow\__init__.py", line 44, in _load
  File "importlib\__init__.py", line 127, in import_module
ModuleNotFoundError: No module named 'tensorflow_core.python'
[7120] Failed to execute script tf_pyinstaller_test
```

**Comment by Legorooj** [comment]
Please post the build log.

**Comment by gcervantes8** [comment]
Ran using: pyinstaller --noupx --additional-hooks-dir=hooks tf_pyinstaller_test.py
```
(temp_venv) C:\Users\Jerry C\Desktop\BUILT>pyinstaller --noupx --additional-hooks-dir=hooks tf_pyinstaller_test.py
124 INFO: PyInstaller: 4.0.dev0
124 INFO: Python: 3.7.8
124 INFO: Platform: Windows-10-10.0.18362-SP0
140 INFO: wrote C:\Users\Jerry C\Desktop\BUILT\tf_pyinstaller_test.spec
140 INFO: UPX is not available.
140 INFO: Extending PYTHONPATH with paths
['C:\\Users\\Jerry C\\Desktop\\BUILT', 'C:\\Users\\Jerry C\\Desktop\\BUILT']
203 INFO: checking Analysis
203 INFO: Building Analysis because Analysis-00.toc is non existent
203 INFO: Initializing module dependency graph...
203 INFO: Caching module graph hooks...
234 INFO: Analyzing base_library.zip ...
3952 INFO: Processing pre-find module path hook distutils from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\pre_find_module_path\\hook-distutils.py'.
3952 INFO: distutils: retargeting to non-venv dir 'C:\\Users\\Jerry C\\AppData\\Local\\Programs\\Python\\Python37\\lib'
6357 INFO: Caching module dependency graph...
6482 INFO: running Analysis Analysis-00.toc
6482 INFO: Adding Microsoft.Windows.Common-Controls to dependent assemblies of final executable
  required by c:\users\jerry c\desktop\built\temp_venv\scripts\python.exe
6498 INFO: Analyzing C:\Users\Jerry C\Desktop\BUILT\tf_pyinstaller_test.py
6592 INFO: Processing pre-find module path hook site from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\pre_find_module_path\\hook-site.py'.
6592 INFO: site: retargeting to fake-dir 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\fake-modules'
12622 INFO: Processing pre-safe import module hook setuptools.extern.six.moves from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\pre_safe_import_module\\hook-setuptools.extern.six.moves.py'.
17277 INFO: Processing pre-safe import module hook six.moves from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\pre_safe_import_module\\hook-six.moves.py'.
19104 INFO: Processing module hooks...
19104 INFO: Loading module hook 'hook-distutils.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
19104 INFO: Loading module hook 'hook-encodings.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
19214 INFO: Loading module hook 'hook-lib2to3.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
19307 INFO: Loading module hook 'hook-numpy.core.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
19464 INFO: Loading module hook 'hook-numpy.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
19464 INFO: Loading module hook 'hook-pkg_resources.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
19885 INFO: Processing pre-safe import module hook win32com from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\pre_safe_import_module\\hook-win32com.py'.
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ModuleNotFoundError: No module named 'win32com'
19979 INFO: Processing pre-safe import module hook win32com from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\pre_safe_import_module\\hook-win32com.py'.
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ModuleNotFoundError: No module named 'win32com'
20292 INFO: Excluding import '__main__'
20292 INFO:   Removing import of __main__ from module pkg_resources
20292 INFO: Loading module hook 'hook-setuptools.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
21104 INFO: Loading module hook 'hook-sysconfig.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
21104 INFO: Loading module hook 'hook-tensorflow.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
21104 INFO: Loading module hook 'hook-xml.dom.domreg.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
21104 INFO: Loading module hook 'hook-xml.etree.cElementTree.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
21104 INFO: Loading module hook 'hook-xml.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
21104 INFO: Loading module hook 'hook-_tkinter.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
21401 INFO: checking Tree
21401 INFO: Building Tree because Tree-00.toc is non existent
21401 INFO: Building Tree Tree-00.toc
21682 INFO: checking Tree
21682 INFO: Building Tree because Tree-01.toc is non existent
21682 INFO: Building Tree Tree-01.toc
21791 INFO: Looking for ctypes DLLs
21901 INFO: Analyzing run-time hooks ...
21901 INFO: Including run-time hook 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\rthooks\\pyi_rth_multiprocessing.py'
21916 INFO: Including run-time hook 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\rthooks\\pyi_rth__tkinter.py'
21932 INFO: Including run-time hook 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\rthooks\\pyi_rth_pkgres.py'
21947 INFO: Looking for dynamic libraries
22869 INFO: Looking for eggs
22869 INFO: Using Python library C:\Users\Jerry C\AppData\Local\Programs\Python\Python37\python37.dll
22869 INFO: Found binding redirects:
[]
22885 INFO: Warnings written to C:\Users\Jerry C\Desktop\BUILT\build\tf_pyinstaller_test\warn-tf_pyinstaller_test.txt
23025 INFO: Graph cross-reference written to C:\Users\Jerry C\Desktop\BUILT\build\tf_pyinstaller_test\xref-tf_pyinstaller_test.html
23103 INFO: checking PYZ
23103 INFO: Building PYZ because PYZ-00.toc is non existent
23103 INFO: Building PYZ (ZlibArchive) C:\Users\Jerry C\Desktop\BUILT\build\tf_pyinstaller_test\PYZ-00.pyz
24931 INFO: Building PYZ (ZlibArchive) C:\Users\Jerry C\Desktop\BUILT\build\tf_pyinstaller_test\PYZ-00.pyz completed successfully.
24962 INFO: checking PKG
24978 INFO: Building PKG because PKG-00.toc is non existent
24978 INFO: Building PKG (CArchive) PKG-00.pkg
25009 INFO: Building PKG (CArchive) PKG-00.pkg completed successfully.
25009 INFO: Bootloader c:\users\jerry c\desktop\built\temp_venv\lib\site-packages\PyInstaller\bootloader\Windows-64bit\run.exe
25009 INFO: checking EXE
25009 INFO: Building EXE because EXE-00.toc is non existent
25009 INFO: Building EXE from EXE-00.toc
25009 INFO: Appending archive to EXE C:\Users\Jerry C\Desktop\BUILT\build\tf_pyinstaller_test\tf_pyinstaller_test.exe
25119 INFO: Building EXE from EXE-00.toc completed successfully.
25134 INFO: checking COLLECT
25134 INFO: Building COLLECT because COLLECT-00.toc is non existent
25134 INFO: Building COLLECT COLLECT-00.toc
35382 INFO: Building COLLECT COLLECT-00.toc completed successfully.
```

Using the --debug all flag:

```
(temp_venv) C:\Users\Jerry C\Desktop\BUILT>pyinstaller --debug all --noupx --additional-hooks-dir=hooks tf_pyinstaller_test.py
93 INFO: PyInstaller: 4.0.dev0
93 INFO: Python: 3.7.8
93 INFO: Platform: Windows-10-10.0.18362-SP0
93 INFO: wrote C:\Users\Jerry C\Desktop\BUILT\tf_pyinstaller_test.spec
93 INFO: UPX is not available.
109 INFO: Extending PYTHONPATH with paths
['C:\\Users\\Jerry C\\Desktop\\BUILT', 'C:\\Users\\Jerry C\\Desktop\\BUILT']
109 INFO: checking Analysis
109 INFO: Building Analysis because Analysis-00.toc is non existent
109 INFO: Initializing module dependency graph...
109 INFO: Caching module graph hooks...
124 INFO: Analyzing base_library.zip ...
2468 INFO: Processing pre-find module path hook distutils from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\pre_find_module_path\\hook-distutils.py'.
2468 INFO: distutils: retargeting to non-venv dir 'C:\\Users\\Jerry C\\AppData\\Local\\Programs\\Python\\Python37\\lib'
3842 INFO: Caching module dependency graph...
3967 INFO: running Analysis Analysis-00.toc
3967 INFO: Adding Microsoft.Windows.Common-Controls to dependent assemblies of final executable
  required by c:\users\jerry c\desktop\built\temp_venv\scripts\python.exe
3983 INFO: Analyzing C:\Users\Jerry C\Desktop\BUILT\tf_pyinstaller_test.py
4045 INFO: Processing pre-find module path hook site from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\pre_find_module_path\\hook-site.py'.
4061 INFO: site: retargeting to fake-dir 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\fake-modules'
7998 INFO: Processing pre-safe import module hook setuptools.extern.six.moves from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\pre_safe_import_module\\hook-setuptools.extern.six.moves.py'.
11153 INFO: Processing pre-safe import module hook six.moves from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\pre_safe_import_module\\hook-six.moves.py'.
12418 INFO: Processing module hooks...
12418 INFO: Loading module hook 'hook-distutils.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
12418 INFO: Loading module hook 'hook-encodings.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
12528 INFO: Loading module hook 'hook-lib2to3.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
12590 INFO: Loading module hook 'hook-numpy.core.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
12731 INFO: Loading module hook 'hook-numpy.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
12731 INFO: Loading module hook 'hook-pkg_resources.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
13059 INFO: Processing pre-safe import module hook win32com from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\pre_safe_import_module\\hook-win32com.py'.
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ModuleNotFoundError: No module named 'win32com'
13153 INFO: Processing pre-safe import module hook win32com from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\pre_safe_import_module\\hook-win32com.py'.
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ModuleNotFoundError: No module named 'win32com'
13481 INFO: Excluding import '__main__'
13481 INFO:   Removing import of __main__ from module pkg_resources
13481 INFO: Loading module hook 'hook-setuptools.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
14215 INFO: Loading module hook 'hook-sysconfig.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
14215 INFO: Loading module hook 'hook-tensorflow.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
14215 INFO: Loading module hook 'hook-xml.dom.domreg.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
14215 INFO: Loading module hook 'hook-xml.etree.cElementTree.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
14215 INFO: Loading module hook 'hook-xml.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
14215 INFO: Loading module hook 'hook-_tkinter.py' from 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks'...
14449 INFO: checking Tree
14449 INFO: Building Tree because Tree-00.toc is non existent
14449 INFO: Building Tree Tree-00.toc
14527 INFO: checking Tree
14527 INFO: Building Tree because Tree-01.toc is non existent
14527 INFO: Building Tree Tree-01.toc
14621 INFO: Looking for ctypes DLLs
14730 INFO: Analyzing run-time hooks ...
14730 INFO: Including run-time hook 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\rthooks\\pyi_rth_multiprocessing.py'
14746 INFO: Including run-time hook 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\rthooks\\pyi_rth__tkinter.py'
14746 INFO: Including run-time hook 'c:\\users\\jerry c\\desktop\\built\\temp_venv\\lib\\site-packages\\PyInstaller\\hooks\\rthooks\\pyi_rth_pkgres.py'
14762 INFO: Looking for dynamic libraries
15605 INFO: Looking for eggs
15605 INFO: Using Python library C:\Users\Jerry C\AppData\Local\Programs\Python\Python37\python37.dll
15605 INFO: Found binding redirects:
[]
Traceback (most recent call last):
  File "C:\Users\Jerry C\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\Jerry C\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\Jerry C\Desktop\BUILT\temp_venv\Scripts\pyinstaller.exe\__main__.py", line 7, in <module>
  File "c:\users\jerry c\desktop\built\temp_venv\lib\site-packages\PyInstaller\__main__.py", line 114, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "c:\users\jerry c\desktop\built\temp_venv\lib\site-packages\PyInstaller\__main__.py", line 65, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "c:\users\jerry c\desktop\built\temp_venv\lib\site-packages\PyInstaller\building\build_main.py", line 716, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "c:\users\jerry c\desktop\built\temp_venv\lib\site-packages\PyInstaller\building\build_main.py", line 663, in build
    exec(code, spec_namespace)
  File "C:\Users\Jerry C\Desktop\BUILT\tf_pyinstaller_test.spec", line 17, in <module>
    noarchive=True)
  File "c:\users\jerry c\desktop\built\temp_venv\lib\site-packages\PyInstaller\building\build_main.py", line 241, in __init__
    self.__postinit__()
  File "c:\users\jerry c\desktop\built\temp_venv\lib\site-packages\PyInstaller\building\datastruct.py", line 160, in __postinit__
    self.assemble()
  File "c:\users\jerry c\desktop\built\temp_venv\lib\site-packages\PyInstaller\building\build_main.py", line 496, in assemble
    for name, path, typecode in compile_py_files(new_toc, CONF['workpath']):
  File "c:\users\jerry c\desktop\built\temp_venv\lib\site-packages\PyInstaller\utils\misc.py", line 152, in compile_py_files
    with open(obj_fnm, 'rb') as fh:
FileNotFoundError: [Errno 2] No such file or directory: '-'
```

**Comment by Legorooj** [comment]
@gcervantes8 I commited a fix for the error in the second log in f7168c866. And `--debug` has no effect on the build log; the output executable is different when using `--debug`. Attempting to reproduce now.

**Comment by Legorooj** [comment]
@vitsensei can you take a look at this?

**Comment by vitsensei** [comment]
@gcervantes8 @joshd-7 I am currently trying to reproduce the bug. However, I found [this ](https://github.com/tensorflow/tensorflow/issues/36441) thread. Apparently tensorflow_core is a new feature in tensorflow 1.15, which explains why @gcervantes8 can create an executable using 1.14.

**Comment by bwoodsend** [comment]
This was fixed in [#46](https://github.com/pyinstaller/pyinstaller-hooks-contrib/pull/46). When we release it you should be able to:
```shell
pip install "pyinstaller-hooks-contrib>=2020.9"
```
But in the meantime use:
```shell
pip install https://github.com/pyinstaller/pyinstaller-hooks-contrib/archive/master.zip
```
